kratos/geometries/triangle_3d_3.h

```diff
@@ -2139,7 +2074,102 @@ template<class TPointType> class Triangle3D3
       return 0.5 * std::sqrt((b+c-a) * (c+a-b) * (a+b-c) / (a+b+c));
     }
 
-	bool ComputeIntervals(double& VV0,
+    bool LineTriangleOverlap(
+        const Point& rPoint1,
+        const Point& rPoint2)
+    {
+        array_1d<double,3> intersection_point;
+        const int result = IntersectionUtilities::ComputeTriangleLineIntersection(*this, rPoint1, rPoint2, intersection_point);
+        if (result == 1) return true;
+        else return false;
+    }
+
+    bool TriangleTriangleOverlap(
+        const Point& rPoint1,
+        const Point& rPoint2,
+        const Point& rPoint3)
+    {
+        // Based on code develop by Moller: http://fileadmin.cs.lth.se/cs/Personal/Tomas_Akenine-Moller/code/opttritri.txt
+        // and the article "A Fast Triangle-Triangle Intersection Test", Journal of Graphics Tools, 2(2), 1997:
+        // http://web.stanford.edu/class/cs277/resources/papers/Moller1997b.pdf
+
+        Plane3D plane_1(this->GetPoint(0), this->GetPoint(1), this->GetPoint(2));
+        array_1d<double, 3> distances_1;
+        distances_1[0] = plane_1.CalculateSignedDistance(rPoint1);
+        distances_1[1] = plane_1.CalculateSignedDistance(rPoint2);
+        distances_1[2] = plane_1.CalculateSignedDistance(rPoint3);
+        if (AllSameSide(distances_1))
+            return false;
+
+        Plane3D plane_2(rPoint1, rPoint2, rPoint3);
+        array_1d<double, 3> distances_2;
+        for (int i = 0; i < 3; i++)

```

Comment:
```suggestion
        for (int i = 0; i < 3; ++i)
```
(very minor, most probably the compiler is changing it)